app/models/post.rb

```diff
@@ -1,5 +1,14 @@
 class Post < ApplicationRecord
   belongs_to :user
-  has_many :comments
-  has_many :likes
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_posts_counter
+    post_counter = user.posts.count
+    user.update(posts_counter: post_counter)
```

Comment:
[OPTIONAL] You could just increment the count after a new post is created instead of counting all the posts:
`author.increment!(:posts_counter)`
Then you call it after saving or after committing:
`after_save :update_post_counter`